x-pack/plugins/beats/server/lib/adapters/database/kibana_database_adapter.ts

```diff
@@ -0,0 +1,112 @@
+ */
+
+import { FrameworkRequest } from '../famework/adapter_types';
+import {
+  DatabaseAdapter,
+  DatabaseBulkIndexDocumentsParams,
+  DatabaseCreateDocumentParams,
+  DatabaseCreateDocumentResponse,
+  DatabaseDeleteDocumentParams,
+  DatabaseDeleteDocumentResponse,
+  DatabaseGetDocumentResponse,
+  DatabaseGetParams,
+  DatabaseIndexDocumentParams,
+  DatabaseKbnESCluster,
+  DatabaseKbnESPlugin,
+  DatabaseMGetParams,
+  DatabaseMGetResponse,
+  DatabasePutTemplateParams,
+  DatabaseSearchParams,
+  DatabaseSearchResponse,
+} from './adapter_types';
+
+export class KibanaDatabaseAdapter implements DatabaseAdapter {
+  private es: DatabaseKbnESCluster;
+
+  constructor(kbnElasticSearch: DatabaseKbnESPlugin) {
+    this.es = kbnElasticSearch.getCluster('admin');
+  }
+  public async putTemplate(
+    req: FrameworkRequest | null,
+    params: DatabasePutTemplateParams
+  ): Promise<any> {
+    const callES = this.getCallType(req);
+    const result = await callES('indices.putTemplate', params);
+    return result;
+  }
+
+  public async get<Source>(
+    req: FrameworkRequest | null,
+    params: DatabaseGetParams
+  ): Promise<DatabaseGetDocumentResponse<Source>> {
+    const callES = this.getCallType(req);
+    const result = await callES('get', params);
+    return result;
+    // todo
+  }
+
+  public async mget<T>(
+    req: FrameworkRequest | null,
+    params: DatabaseMGetParams
+  ): Promise<DatabaseMGetResponse<T>> {
+    const callES = this.getCallType(req);
+    const result = await callES('mget', params);
+    return result;
+    // todo
+  }
+
+  public async bulk(
+    req: FrameworkRequest | null,
+    params: DatabaseBulkIndexDocumentsParams
+  ): Promise<any> {
+    const callES = this.getCallType(req);
+    const result = await callES('bulk', params);
+    return result;
+  }
+
+  public async create(
+    req: FrameworkRequest | null,
+    params: DatabaseCreateDocumentParams
+  ): Promise<DatabaseCreateDocumentResponse> {
+    const callES = this.getCallType(req);
+    const result = await callES('create', params);
+    return result;
+  }
+  public async index<T>(
+    req: FrameworkRequest | null,
+    params: DatabaseIndexDocumentParams<T>
+  ): Promise<any> {
+    const callES = this.getCallType(req);
+    const result = await callES('index', params);
+    return result;
+  }
+  public async delete(
+    req: FrameworkRequest | null,
+    params: DatabaseDeleteDocumentParams
+  ): Promise<DatabaseDeleteDocumentResponse> {
+    const callES = this.getCallType(req);
+    const result = await callES('delete', params);
+    return result;
+  }
+
+  public async search<Source>(
+    req: FrameworkRequest | null,
+    params: DatabaseSearchParams
+  ): Promise<DatabaseSearchResponse<Source>> {
+    const callES = this.getCallType(req);
+    const result = await callES('search', params);
+    return result;
+  }
+
+  private getCallType(req: FrameworkRequest | null): any {
```

Comment:
I did this because the typing on data returned from callWithRequest is assumed, and in fact at runtime could be very incorrect. This framework adapter is only used by other domain based adapters such as a beats or tags adapter. Those other adapters primary action is to conform the returned data and return it neatly. Thus expecting any from this (or unknown in future versions of TS) then properly formatting at runtime and returning an accurate type is IMHO a more "honest" approach to typing this solution.

Reply:
Yes, good point. I wonder if `elasticsearch-js` performs any validation. :thinking:

Reply:
Not so far as I can tell